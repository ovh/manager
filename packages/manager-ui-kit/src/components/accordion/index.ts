export { Accordion } from './Accordion.component';
export type { AccordionProps } from './Accordion.props';

export { AccordionContent } from './accordion-content/AccordionContent.component';
export { AccordionItem } from './accordion-item/AccordionItem.component';
export { AccordionTrigger } from './accordion-trigger/AccordionTrigger.component';
