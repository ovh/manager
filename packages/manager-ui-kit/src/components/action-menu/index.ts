export { ActionMenu } from './ActionMenu.component';
export type { ActionMenuItem, ActionMenuProps } from './ActionMenu.props';
