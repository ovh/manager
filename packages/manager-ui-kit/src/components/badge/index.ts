export { Badge } from './badge.component';

export type { BadgeProps } from './badge.props';

export * from './Badge.constants';
