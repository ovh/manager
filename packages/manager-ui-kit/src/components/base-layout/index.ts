export { BaseLayout } from './BaseLayout.component';

export type { BaseLayoutProps } from './BaseLayout.props';

export type { HeaderProps } from './header/Header.props';
