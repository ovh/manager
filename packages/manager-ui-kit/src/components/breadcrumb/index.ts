export { Breadcrumb } from './Breadcrumb.component';
export type { BreadcrumbProps } from './Breadcrumb.props';
export * from './Breadcrumb.constants';
