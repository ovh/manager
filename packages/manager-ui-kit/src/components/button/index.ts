export { Button } from './Button.component';
export type { ButtonProps } from './Button.props';
