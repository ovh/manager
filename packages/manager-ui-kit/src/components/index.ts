export * from './action-banner';
export * from './action-menu';
export * from './redirection-guard';
export * from './breadcrumb';
export * from './base-layout';
export * from './clipboard';
export * from './step';
export * from './tabs';
export * from './tiles-input-group';
export * from './tiles-input';

export * from './Link';
export * from './drawer';
export * from './guide-menu';

export * from './notifications';

export * from './price';

export * from './filters';

export * from './button';
export * from './text';

export * from './service-state-badge';

export * from './tile';
export * from './order';

export * from './badge';
export * from './modal';
export * from './tags-list';
export * from './tags-tile';
export * from './changelog-menu';
export * from './link-card';
export * from './error';
export * from './error-boundary';
export * from './update-name-modal';
export * from './delete-modal';
export * from './grid-layout';
export * from './onboarding-layout';
export * from './datagrid';
export * from './accordion';
export * from './checkbox';
export * from './combobox';
export * from './datepicker';
export * from './file-upload';
export * from './form-field';
export * from './message';
export * from './phone-number';
export * from './popover';
export * from './progress';
export * from './quantity';
export * from './radio-group';
export * from './select';
export * from './switch';
export * from './timepicker';
export * from './toggle';
export * from './tooltip';
export * from './tree-view';
