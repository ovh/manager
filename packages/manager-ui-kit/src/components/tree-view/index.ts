export { TreeView } from './TreeView.component';
export type { TreeViewProps } from './TreeView.props';

export { TreeViewNode } from './tree-view-node/TreeViewNode.component';
export { TreeViewNodes } from './tree-view-nodes/TreeViewNodes.component';
