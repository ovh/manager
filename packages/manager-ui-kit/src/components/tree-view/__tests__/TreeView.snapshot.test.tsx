import { render } from '@testing-library/react';
import { describe, expect, it } from 'vitest';

import { TreeView, TreeViewNode, TreeViewNodes } from '@/components';

describe('TreeView Snapshot tests', () => {
  it('renders the component with default props and children', () => {
    const { container } = render(
      <TreeView
        items={[
          {
            children: [
              {
                id: 'app.tsx',
                name: 'app.tsx',
              },
              {
                id: 'index.ts',
                name: 'index.ts',
              },
              {
                children: [
                  {
                    id: 'Button.tsx',
                    name: 'Button.tsx',
                  },
                  {
                    id: 'Card.tsx',
                    name: 'Card.tsx',
                  },
                ],
                id: 'components',
                name: 'components',
              },
            ],
            id: 'src',
            name: 'src',
          },
          {
            id: 'package.json',
            name: 'package.json',
          },
          {
            id: 'readme.md',
            name: 'README.md',
          },
        ]}
      >
        <TreeViewNodes>
          <TreeViewNode
            item={{
              children: [
                {
                  id: 'app.tsx',
                  name: 'app.tsx',
                },
                {
                  id: 'index.ts',
                  name: 'index.ts',
                },
                {
                  children: [
                    {
                      id: 'Button.tsx',
                      name: 'Button.tsx',
                    },
                    {
                      id: 'Card.tsx',
                      name: 'Card.tsx',
                    },
                  ],
                  id: 'components',
                  name: 'components',
                },
              ],
              id: 'src',
              name: 'src',
            }}
          />
          <TreeViewNode
            item={{
              id: 'package.json',
              name: 'package.json',
            }}
          />
          <TreeViewNode
            item={{
              id: 'readme.md',
              name: 'README.md',
            }}
          />
        </TreeViewNodes>
      </TreeView>,
    );
    expect(container).toMatchSnapshot();
  });
});
