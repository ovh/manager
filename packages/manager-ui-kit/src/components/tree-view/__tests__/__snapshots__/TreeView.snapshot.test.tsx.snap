// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`TreeView Snapshot tests > renders the component with default props and children 1`] = `
<div>
  <div
    class="_tree-view_1dn67_2"
    data-ods="tree-view"
    data-part="root"
    data-scope="tree-view"
    dir="ltr"
    id="tree::r0::root"
  >
    <div
      aria-label="Tree View"
      aria-labelledby="tree::r0::label"
      data-part="tree"
      data-scope="tree-view"
      dir="ltr"
      id="tree::r0::tree"
      role="tree"
      tabindex="-1"
    >
      <div
        class="_tree-view-nodes_1hk4c_2"
        data-ods="tree-view-nodes"
      >
        <div
          aria-disabled="false"
          aria-expanded="false"
          aria-level="1"
          aria-selected="false"
          data-branch="src"
          data-depth="1"
          data-ods="tree-view-node-branch"
          data-ownedby="tree::r0::tree"
          data-part="branch"
          data-path="0"
          data-scope="tree-view"
          data-state="closed"
          data-value="src"
          dir="ltr"
          id="collapsible::r1:"
          role="treeitem"
          style="--depth: 1;"
        >
          <div
            aria-disabled="false"
            class="_tree-view-node-branch__control_q2e1u_2"
            data-depth="1"
            data-focus=""
            data-part="branch-control"
            data-path="0"
            data-scope="tree-view"
            data-state="closed"
            data-value="src"
            dir="ltr"
            id="tree::r0::node:src"
            role="group"
            tabindex="0"
          >
            <div
              aria-hidden="true"
              class="_tree-view-node-branch__control__chevron_q2e1u_42"
              data-focus=""
              data-part="branch-indicator"
              data-scope="tree-view"
              data-state="closed"
            >
              <span
                class="_icon_10c23_2 _icon--chevron-right_10c23_97"
                data-ods="icon"
                role="presentation"
              />
            </div>
            <span
              data-part="branch-text"
              data-scope="tree-view"
              data-state="closed"
              dir="ltr"
            >
              src
            </span>
          </div>
          <div
            class="_tree-view-node-branch__control__children_q2e1u_54"
            data-collapsible=""
            data-depth="1"
            data-part="branch-content"
            data-path="0"
            data-scope="tree-view"
            data-state="closed"
            data-value="src"
            dir="ltr"
            hidden=""
            id="collapsible::r1::content"
            role="group"
            style="--height: 0px; --width: 0px;"
          >
            <div
              data-depth="1"
              data-part="branch-indent-guide"
              data-scope="tree-view"
            />
            <div
              aria-disabled="false"
              aria-level="2"
              aria-selected="false"
              class="_tree-view-node-item_1ktxc_2"
              data-depth="2"
              data-ods="tree-view-node-item"
              data-ownedby="tree::r0::tree"
              data-part="item"
              data-path="0/0"
              data-scope="tree-view"
              data-value="app.tsx"
              dir="ltr"
              id="tree::r0::node:app.tsx"
              role="treeitem"
              style="--depth: 2;"
              tabindex="-1"
            >
              <span
                data-part="item-text"
                data-scope="tree-view"
              >
                app.tsx
              </span>
            </div>
            <div
              aria-disabled="false"
              aria-level="2"
              aria-selected="false"
              class="_tree-view-node-item_1ktxc_2"
              data-depth="2"
              data-ods="tree-view-node-item"
              data-ownedby="tree::r0::tree"
              data-part="item"
              data-path="0/1"
              data-scope="tree-view"
              data-value="index.ts"
              dir="ltr"
              id="tree::r0::node:index.ts"
              role="treeitem"
              style="--depth: 2;"
              tabindex="-1"
            >
              <span
                data-part="item-text"
                data-scope="tree-view"
              >
                index.ts
              </span>
            </div>
            <div
              aria-disabled="false"
              aria-expanded="false"
              aria-level="2"
              aria-selected="false"
              data-branch="components"
              data-depth="2"
              data-ods="tree-view-node-branch"
              data-ownedby="tree::r0::tree"
              data-part="branch"
              data-path="0/2"
              data-scope="tree-view"
              data-state="closed"
              data-value="components"
              dir="ltr"
              id="collapsible::r2:"
              role="treeitem"
              style="--depth: 2;"
            >
              <div
                aria-disabled="false"
                class="_tree-view-node-branch__control_q2e1u_2"
                data-depth="2"
                data-part="branch-control"
                data-path="0/2"
                data-scope="tree-view"
                data-state="closed"
                data-value="components"
                dir="ltr"
                id="tree::r0::node:components"
                role="group"
                tabindex="-1"
              >
                <div
                  aria-hidden="true"
                  class="_tree-view-node-branch__control__chevron_q2e1u_42"
                  data-part="branch-indicator"
                  data-scope="tree-view"
                  data-state="closed"
                >
                  <span
                    class="_icon_10c23_2 _icon--chevron-right_10c23_97"
                    data-ods="icon"
                    role="presentation"
                  />
                </div>
                <span
                  data-part="branch-text"
                  data-scope="tree-view"
                  data-state="closed"
                  dir="ltr"
                >
                  components
                </span>
              </div>
              <div
                class="_tree-view-node-branch__control__children_q2e1u_54"
                data-collapsible=""
                data-depth="2"
                data-part="branch-content"
                data-path="0/2"
                data-scope="tree-view"
                data-state="closed"
                data-value="components"
                dir="ltr"
                hidden=""
                id="collapsible::r2::content"
                role="group"
                style="--height: 0px; --width: 0px;"
              >
                <div
                  data-depth="2"
                  data-part="branch-indent-guide"
                  data-scope="tree-view"
                />
                <div
                  aria-disabled="false"
                  aria-level="3"
                  aria-selected="false"
                  class="_tree-view-node-item_1ktxc_2"
                  data-depth="3"
                  data-ods="tree-view-node-item"
                  data-ownedby="tree::r0::tree"
                  data-part="item"
                  data-path="0/2/0"
                  data-scope="tree-view"
                  data-value="Button.tsx"
                  dir="ltr"
                  id="tree::r0::node:Button.tsx"
                  role="treeitem"
                  style="--depth: 3;"
                  tabindex="-1"
                >
                  <span
                    data-part="item-text"
                    data-scope="tree-view"
                  >
                    Button.tsx
                  </span>
                </div>
                <div
                  aria-disabled="false"
                  aria-level="3"
                  aria-selected="false"
                  class="_tree-view-node-item_1ktxc_2"
                  data-depth="3"
                  data-ods="tree-view-node-item"
                  data-ownedby="tree::r0::tree"
                  data-part="item"
                  data-path="0/2/1"
                  data-scope="tree-view"
                  data-value="Card.tsx"
                  dir="ltr"
                  id="tree::r0::node:Card.tsx"
                  role="treeitem"
                  style="--depth: 3;"
                  tabindex="-1"
                >
                  <span
                    data-part="item-text"
                    data-scope="tree-view"
                  >
                    Card.tsx
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          aria-disabled="false"
          aria-level="1"
          aria-selected="false"
          class="_tree-view-node-item_1ktxc_2"
          data-depth="1"
          data-ods="tree-view-node-item"
          data-ownedby="tree::r0::tree"
          data-part="item"
          data-path="1"
          data-scope="tree-view"
          data-value="package.json"
          dir="ltr"
          id="tree::r0::node:package.json"
          role="treeitem"
          style="--depth: 1;"
          tabindex="-1"
        >
          <span
            data-part="item-text"
            data-scope="tree-view"
          >
            package.json
          </span>
        </div>
        <div
          aria-disabled="false"
          aria-level="1"
          aria-selected="false"
          class="_tree-view-node-item_1ktxc_2"
          data-depth="1"
          data-ods="tree-view-node-item"
          data-ownedby="tree::r0::tree"
          data-part="item"
          data-path="2"
          data-scope="tree-view"
          data-value="readme.md"
          dir="ltr"
          id="tree::r0::node:readme.md"
          role="treeitem"
          style="--depth: 1;"
          tabindex="-1"
        >
          <span
            data-part="item-text"
            data-scope="tree-view"
          >
            README.md
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;
